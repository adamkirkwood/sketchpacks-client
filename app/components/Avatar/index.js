import React from 'react'

const Avatar = ({ handle, width, height }) => (
  <div>
    <img className="image is-maker"
      src={`https://github.com/${handle}.png`}
      role="presentation"
      width={width}
      height={height}
    />
  </div>
)

Avatar.defaultProps = { width: 24, height: 24 }

export default Avatar
