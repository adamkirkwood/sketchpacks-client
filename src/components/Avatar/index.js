import React from 'react'
import LazyLoad from 'react-lazyload'

const Avatar = ({ handle, width, height }) => (
  <div>
    <LazyLoad height={height} offset={200} unmountIfInvisible debounce={500}>
      <img className="image is-maker"
        src={`https://github.com/${handle}.png`}
        role="presentation"
        width={width}
        height={height}
      />
    </LazyLoad>
  </div>
)

Avatar.defaultProps = { width: 24, height: 24 }

export default Avatar
