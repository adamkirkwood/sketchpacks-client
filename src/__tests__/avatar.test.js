import React from 'react'
import Avatar from '../components/Avatar'

describe('Avatar', () => {
  it('should render an avatar', () => {
    const wrapper = shallow(
        <Avatar handle={'sketchpacks'} width={24} height={24} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
