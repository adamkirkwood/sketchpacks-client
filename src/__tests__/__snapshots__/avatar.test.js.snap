// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render an avatar 1`] = `
<div>
  <img
    className="image is-maker"
    height={24}
    role="presentation"
    src="https://github.com/sketchpacks.png"
    width={24}
  />
</div>
`;
